Use of non-temporary ipv6 address

aravindprasads opened the pull request:
Fix for https://github.com/radcli/radcli/issues/23 

Issue Description:
In Linux,/BSD/Apple-OS, it is possible to enable temporary IPv6 addresses in system by default (when IPv6 Privacy Extensions are enabled in system). In this scenario, with default settings (if bindaddr option is set
to "*" in config file), the radius client will use the temporary ipv6 address as source address for the packets sent out. This is usually not desired since the RADIUS Server identifies the RADIUS Client usually by a unique address and continuously changing IPV6 addresses requires changes in RADIUS Server config too which is not preferable.

Fix:
In these cases, it is preferable to make Radius Client to use Global IPv6 address as source address for packets sent out.

Changes made:
Introduced a config file option to specify support of IPV6 non-temporary address support. If this option is set to "true" in config file, Radius Client will use Global IPv6 address as source address for packets sent out.

aravindprasads committed:
Support for non-temporary IPv6 address

Description:
In Linux/BSD/Apple-OS, it is possible to enable temporary IPv6 addresses in
system by default (when IPv6 Privacy Extensions are enabled in system). In this
scenario, with default settings (if bindaddr option is set to "*" in config
file), the radius client will use the temporary ipv6 address as source address
for the packets sent out. This is usually not desired since the RADIUS Server
identifies the RADIUS Client usually by a unique address and continuously
changing IPV6 addresses requires changes in RADIUS Server config too which is
not preferable.

Fix:
In these cases, it is preferable to make Radius Client to use Global IPv6
address as source address for packets sent out.

Changes made:
Introduced a config file option to specify support of IPV6 non-temporary
address support. If this option is set to "true" in config file, Radius Client
will use Global IPv6 address as source address for packets sent out.

aravindprasads commented:
Hi @nmav @hzuleger 

Kindly review the changes and let me know your views.

aravindprasads commented:
Hi @nmav 
Kindly review the changes and let me know your views.

nmav commented on etc/radiusclient-tls.conf.in:
```diff
@@ -57,6 +57,14 @@ bindaddr	*
+# IPv6 Global address and will not use the temporary adresses. If commented 
+# out, temporary IPv6 addresses will be used as source address for the packets
+# sent.
+#ipv6_use_non_temp_addr	true
```
shouldn't we have the existing behavior being the default?

nmav commented on etc/radiusclient.conf.in:
```diff
@@ -62,5 +62,13 @@ bindaddr	*
+# IPv6 Global address and will not use the temporary adresses. If commented                         
+# out, temporary IPv6 addresses will be used as source address for the packets                      
+# sent.                                                                                             
+#ipv6_use_non_temp_addr	true                                                                        
```
same here

nmav commented on lib/options.h:
```diff
@@ -33,6 +33,7 @@ static OPTION config_options_default[] = {
 {"serv-type",		OT_STR, ST_UNDEF, NULL},
 {"serv-auth-type",	OT_STR, ST_UNDEF, NULL}, /* alias for serv-type */
 {"namespace",		OT_STR, ST_UNDEF, NULL}, 
+{"ipv6_use_non_temp_addr",	OT_STR, ST_UNDEF, NULL}, 
```
maybe we can get rid of the ipv6 prefix, and use hyphens instead of underscores; simply `use-non-temp-addr`. Then we only document that this is an ipv6-only option.

nmav commented on tests/ipv6-non-temp-addr-tests.sh:
```diff
@@ -0,0 +1,73 @@
+echo "ipv6_use_non_temp_addr	true" >> radiusclient-temp$PID.conf
+
+#enable ipv6 privay settings
+sudo sysctl -w net.ipv6.conf.all.use_tempaddr=2
```
I do not think we need to use sudo in the script; travis already runs  as root, and it is not a good practice to call sudo in a test suite. We can use:
```
        if test "$(id -u)" != "0";then
                exit 77
        fi
```
in the beginning of the script to ensure that this will be skipped if the testsuite is run as root.

nmav reviewed (changes requested):
Overall it is fine except for the comments above.

aravindprasads commented:
Hi @nmav 

Thanks for the review.
1) Removed ipv6 prefix and replaced underscores with hyphens for the option. 
2) Added sudo in travis.yml file since the tests were not run with root access
2) Added checks in test file for root access and removed sudo usage in the test file.

>>> shouldn't we have the existing behavior being the default?

You mean, should this option be always present in conf file (should we remove the # for the option in conf files) ??

nmav commented:
>>  shouldn't we have the existing behavior being the default?
> You mean, should this option be always present in conf file (should we remove the # for the option in conf files) ??

No sorry for the confusion. May I request a last change? The 'use-non-temp-addr' is quite confusing to me due to the negation. Why not avoid the negation and have an option 'use-temp-addr' or 'use-public-addr'?

aravindprasads commented:
Hi @nmav 

Thanks for the review. Changed the option name to "use-public-addr".  

Kindly review the same. Thanks for the support.

aravindprasads commented:
Hi @nmav 

Kindly review the changes and let me know if changes are fine. Thanks for the support.

nmav commented:
Merged. Thank you!
